test/client/unit/specs/components/avatar.vue.test.js

```diff
@@ -0,0 +1,327 @@
+    beforeEach(() => {
+      vm.member.preferences = { background: 'pony' };
+    });
+
+    it('shows the background', () => {
+      expect(vm.backgroundClass).to.equal('background_pony');
+    });
+
+    it('can be overridden', () => {
+      vm.overrideAvatarGear = { background: 'character' };
+
+      expect(vm.backgroundClass).to.equal('background_character');
+    });
+
+    it('returns to a blank string if not showing background', () => {
+      vm.withBackground = false;
+      vm.avatarOnly = true;
+
+      expect(vm.backgroundClass).to.equal('');
+    });
+  });
+
+  describe('specialMountClass', () => {
+    it('checks if riding a Kangaroo', () => {
+      vm.member = {
+        stats: {
+          class: 'None',
+        },
+        items: {},
+      };
+
+      expect(vm.specialMountClass).to.equal(undefined);
+
+      vm.member.items = {
+        currentMount: ['Kangaroo'],
+      };
+
+      expect(vm.specialMountClass).to.equal('offset-kangaroo');
+    });
+  });
+
+  describe('skinClass', () => {
+    it('returns current skin color', () => {
+      vm.member = {
+        stats: {},
+        preferences: {
+          skin: 'blue',
+        },
+      };
+
+      expect(vm.skinClass).to.equal('skin_blue');
+    });
+
+    it('returns if sleep or not', () => {
+      vm.member = {
+        stats: {},
+        preferences: {
+          skin: 'blue',
+          sleep: false,
+        },
+      };
+
+      expect(vm.skinClass).to.equal('skin_blue');
+
+      vm.member.preferences.sleep = true;
+
+      expect(vm.skinClass).to.equal('skin_blue_sleep');
+    });
+  });
+
+  context('methods', () => {
+    describe('getGearClass', () => {
+      beforeEach(() => {
+        vm.member = {
+          items: {
+            gear: {
+              equipped: { Hat: 'Fancy Tophat' },
+            },
+          },
+          preferences: { costume: false },
+        };
+      });
+
+      it('returns undefined if no match', () => {
+        expect(vm.getGearClass('foo')).to.equal(undefined);
+      });
+
+      it('returns the matching gear', () => {
+        expect(vm.getGearClass('Hat')).to.equal('Fancy Tophat');
+      });
+
+      it('can be overridden', () => {
+        vm.overrideAvatarGear = { Hat: 'Dapper Bowler' };
+
+        expect(vm.getGearClass('Hat')).to.equal('Dapper Bowler');
+      });
+    });
+
+    describe('hideGear', () => {
+      it('returns no weapon equipped', () => {
+        vm.member.items.gear.equipped = {};
+        expect(vm.hideGear('weapon')).to.equal(false);
+      });
+
+      beforeEach(() => {
+        vm.member = {
+          items: {
+            gear: {
+              equipped: {
+                weapon: {
+                  baseWeapon: 'Spoon',
+                  twoHanded: false,
+                },
+              },
+            },
+          },
+          preferences: { costume: false },
+        };
+      });
+
+      xit('returns false if not two handed weapon', () => {
```

Comment:
can you add a comment on why these are disabled? or if they can't be fixed, can you remove them? Thanks